### PR TITLE
[ADD] stock_landed_costs_security: New user group to manage landed costs

### PR DESCRIPTION
**Pull request by cesar-tecnativa**
TT37364

### CODE REVIEW

**Comment by cesar-tecnativa** [comment]
@pedrobaeza Please could you review?

**Review by rousseldenis — APPROVED** [review]
Code review

File: stock_landed_costs_security/readme/DESCRIPTION.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+This addons adds the Landing cost user group. Any user mapped to this grou can create an validate landing costs.
```
Comment (pedrobaeza) [inline stock_landed_costs_security/readme/DESCRIPTION.rst]:
```suggestion
This addon adds the "Landing cost" user group. Any user mapped to this group can create and validate landing costs.
```

File: stock_landed_costs_security/readme/USAGE.rst [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#. Go to Users.
+#. Select Landing cost -> User.
```
Comment (pedrobaeza) [inline stock_landed_costs_security/readme/USAGE.rst]:
```suggestion
#. Select Landing costs -> User.
```

And you can put these two first lines in CONFIGURE.rst

File: stock_landed_costs_security/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+#. Go to Inventory > Operations > Landed Costs.
+#. This user can create and validate any landed costs.
```
Comment (pedrobaeza) [inline stock_landed_costs_security/readme/USAGE.rst]:
```suggestion
#. This user can create and validate any landed cost.
```

File: stock_landed_costs_security/readme/USAGE.rst [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#. Go to Users.
+#. Select Landing cost -> User.
+#. Go to Inventory > Operations > Landed Costs.
```
Comment (pedrobaeza) [inline stock_landed_costs_security/readme/USAGE.rst]: There should be a new top level menu entry. Better to point to it.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
I'm missing the top level menu and the SVG file

File: stock_landed_costs_security/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2022 Tecnativa - César A. Sánchez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+{
+    "name": "Stock landed costs security",
+    "version": "13.0.1.0.0",
+    "category": "Inventory",
+    "website": "https://github.com/OCA/stock-logistics-workflow",
+    "author": "Tecnativa, Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "depends": ["stock_landed_costs", "account", "stock_account"],
```
Comment (pedrobaeza) [inline stock_landed_costs_security/__manifest__.py]: Why not depending only in `stock_landed_costs`? The rest are already dependencies of this one.

File: stock_landed_costs_security/readme/DESCRIPTION.rst [hunk 6]
```diff
@@ -0,0 +1 @@
+This addon adds the "Landing cost" user group. Any user mapped to this group can create and validate landing costs.
```
Comment (pedrobaeza) [inline stock_landed_costs_security/readme/DESCRIPTION.rst]:
```suggestion
This module promotes the "Landing cost" feature to a full application, with its icon and specific menu, accessible through an specific user group. Any user mapped to this group can create and validate landing costs without extra permissions.
```

File: stock_landed_costs_security/readme/USAGE.rst [hunk 7]
```diff
@@ -0,0 +1,4 @@
+#. Go to Users.
+#. Select Landing costs -> User.
```
Comment (pedrobaeza) [inline stock_landed_costs_security/readme/USAGE.rst:2]: Put these 2 steps in CONFIGURE.rst, and rephrase it this way:

```suggestion
#. Go to *Settings > Users and companies > Users.
#. Select one user or create a new one.
#. Select on the section "Operations", line "Landing costs", the option "User".
```

File: stock_landed_costs_security/readme/USAGE.rst [hunk 8]
```diff
@@ -0,0 +1,4 @@
...
+#. Go to Inventory > Operations > Landed Costs or Go to Landed Costs.
```
Comment (pedrobaeza) [inline stock_landed_costs_security/readme/USAGE.rst]:
```suggestion
#. Go to *Landed Costs*.
```

File: stock_landed_costs_security/security/ir.model.access.csv [hunk 9]
```diff
@@ -0,0 +1,7 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_stock_landed_cost_security,stock.landed.cost,stock_landed_costs.model_stock_landed_cost,stock_landed_costs_security.group_stock_landed_costs_security_user,1,1,1,1
+access_stock_landed_cost_lines_security,stock.landed.cost.lines,stock_landed_costs.model_stock_landed_cost_lines,stock_landed_costs_security.group_stock_landed_costs_security_user,1,1,1,1
+access_stock_valuation_adjustment_lines_security,stock.valuation.adjustment.lines,stock_landed_costs.model_stock_valuation_adjustment_lines,stock_landed_costs_security.group_stock_landed_costs_security_user,1,1,1,1
+access_account_journal_landed_cost_security,account.journal,account.model_account_journal,stock_landed_costs_security.group_stock_landed_costs_security_user,1,0,0,0
```
Comment (pedrobaeza) [inline stock_landed_costs_security/security/ir.model.access.csv:5]: Why do you need these permissions?

File: stock_landed_costs_security/security/stock_landed_costs_security.xml [hunk 10]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="0">
+    <record id="stock_landed_costs_security_category" model="ir.module.category">
+        <field name="name">Landing cost</field>
+        <field name="sequence">7</field>
+    </record>
+    <record id="group_stock_landed_costs_security_user" model="res.groups">
+        <field name="name">User</field>
+        <field name="category_id" ref="stock_landed_costs_security_category" />
+        <field name="users" eval="[(4, ref('base.user_root'))]" />
```
Comment (pedrobaeza) [inline stock_landed_costs_security/security/stock_landed_costs_security.xml]:
```suggestion
        <field name="users" eval="[(4, ref('base.user_root')), (4, ref('base.user_admin'))]"/>
```

File: stock_landed_costs_security/security/stock_landed_costs_security.xml [hunk 11]
```diff
@@ -0,0 +1,13 @@
...
+        <field name="implied_ids" eval="[(4, ref('stock.group_stock_user'))]" />
```
Comment (pedrobaeza) [inline stock_landed_costs_security/security/stock_landed_costs_security.xml]: This is not correct. We want to be totally isolated.

```suggestion
```

File: stock_landed_costs_security/security/stock_landed_costs_security.xml [hunk 12]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="0">
+    <record id="stock_landed_costs_security_category" model="ir.module.category">
+        <field name="name">Landing cost</field>
```
Comment (pedrobaeza) [inline stock_landed_costs_security/security/stock_landed_costs_security.xml]: Proper spelling + put it on the "Operations" section (the same as Inventory):

```suggestion
        <field name="name">Landing costs</field>
        <field name="parent_id" ref="base.module_category_operations" />
```

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The SVG file should be included as well on the same folder than the PNG

File: README.md [hunk 13]
```diff
@@ -1,111 +0,0 @@
-
```
Comment (pedrobaeza) [inline README.md]: Please restore this file

File: stock_landed_costs_security/readme/USAGE.rst (hunk 7 above)
Comment (pedrobaeza) [reply stock_landed_costs_security/readme/USAGE.rst:2]: This is still not changed

File: stock_landed_costs_security/security/stock_landed_costs_security.xml [hunk 14]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo noupdate="0">
+    <record id="stock_landed_costs_security_category" model="ir.module.category">
+        <field name="name">Landing costs</field>
+        <field name="sequence">7</field>
```
Comment (pedrobaeza) [inline stock_landed_costs_security/security/stock_landed_costs_security.xml:6]: Still missing the `parent_id`

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The SVG is still missing

File: stock_landed_costs_security/readme/USAGE.rst (hunk 7 above)
Comment (cesar-tecnativa) [reply stock_landed_costs_security/readme/USAGE.rst:2]: it is done

File: stock_landed_costs_security/security/stock_landed_costs_security.xml (hunk 14 above)
Comment (cesar-tecnativa) [reply stock_landed_costs_security/security/stock_landed_costs_security.xml:6]: I didn't see in your previous comments the parent_id.
which one should be?

File: README.md (hunk 13 above)
Comment (cesar-tecnativa) [reply README.md]: It is restored

File: stock_landed_costs_security/security/stock_landed_costs_security.xml (hunk 14 above)
Comment (pedrobaeza) [reply stock_landed_costs_security/security/stock_landed_costs_security.xml:6]: Please check this comment: https://github.com/OCA/stock-logistics-workflow/pull/1017#discussion_r908864135

File: README.md (hunk 13 above)
Comment (pedrobaeza) [reply README.md]: I'm afraid not. This one is the README at directory top level. You have deleted it (probably by mistake).

File: stock_landed_costs_security/security/stock_landed_costs_security.xml (hunk 14 above)
Comment (pedrobaeza) [reply stock_landed_costs_security/security/stock_landed_costs_security.xml:6]: This one is still pending

**Review by pedrobaeza — APPROVED** [review]

**Review by HaraldPanten — APPROVED** [review]

**Pull request merged**